### ISSUE TITLE
Data loss with --replay and hard-linked files due to missing checksums.

### ISSUE THREAD

**Issue by misieck** [sahib/rmlint#672]
Under certain conditions, rmlint does not calculate checksums of encountered files. **All entries without checksums are seemingly interpreted by --replay as one and the same file, irrelevant of their content. This results in deletion of all but one of such files.** 
One such condition that I found is when encountering hard links. Consider the following test setup:

```
\testdir
    - file_one 
    - file_one_link
    - file_two 
    - file_two_link
```

Sizes of file_one and file_two must differ. Running rmlint produces a rmlint.json file with no checksums in it, but the actions are correct.

```
# Duplicate(s):
    ls '/home/muser/rmlint/test/two'
    rm '/home/muser/rmlint/test/two_link'
    ls '/home/muser/rmlint/test/one'
    rm '/home/muser/rmlint/test/one_link'

```
Running rmlint --replay rmlint.json afterwards produces a script attempting to delete all but one of the listed files:
```
# Duplicate(s):
    ls '/home/muser/rmlint/test/one'
    rm '/home/muser/rmlint/test/one_link'
    rm '/home/muser/rmlint/test/two'
    rm '/home/muser/rmlint/test/two_link'
```

This would delete both two and two_link.

**Comment by vassilit** [comment]
I reproduce with `version 2.10.2 compiled: Jan  3 2025 at [14:24:47] "Ludicrous Lemur" (rev f0ca57ec)`

**Comment by StatusCode404** [comment]
It is in the man page for "--replay", well I clearly states that it won't do the default  "--hardlinked"...
```
By design, some options will not have any effect. Those are:

    --followlinks
    --algorithm
    --paranoid
    --clamp-low
    --hardlinked
    --write-unfinished
    … and all other caching options below.
```
https://rmlint.readthedocs.io/en/master/rmlint.1.html

**Comment by StatusCode404** [comment]
This is not a bug since it is by design.
This should go under a feature request for the next version.

**Comment by misieck** [comment]
No,  --hardlinked, means report hard links as duplicates. The negative of that is not to report them as duplicates and thus preserve both files.
 *rmlint deletes both files*. There is no world this could be by design. People are losing data with this.

**Comment by vassilit** [comment]
Indeed, if rmlint would only delete all but one reference to the same inode, the only lost information would be the file entry in those directories, so basically the name, some attributes and path. This is by design.

Here rmlint deletes all references to the same inode, so you lose the content of the file(s).
